bookkeeper-server/src/main/java/org/apache/bookkeeper/server/Main.java

```diff
@@ -327,13 +328,18 @@ static LifecycleComponentStack buildBookieServer(BookieConfiguration conf) throw
         // 5. build extra services
         String[] extraComponents = conf.getServerConf().getExtraServerComponents();
         if (null != extraComponents) {
-            List<ServerLifecycleComponent> components = loadServerComponents(
-                extraComponents,
-                conf,
-                rootStatsLogger);
-            for (ServerLifecycleComponent component : components) {
-                serverBuilder.addComponent(component);
-                log.info("Load lifecycle component : {}", component.getClass().getName());
+            try {
+                List<ServerLifecycleComponent> components = loadServerComponents(
+                    extraComponents,
+                    conf,
+                    rootStatsLogger);
+                for (ServerLifecycleComponent component : components) {
+                    serverBuilder.addComponent(component);
+                    log.info("Load lifecycle component : {}", component.getClass().getName());
+                }
+            } catch (Exception e) {
+                log.info("Failed to load extra components '{}' - {}. Continuing without those components.",
+                    StringUtils.join(extraComponents), e.getMessage());
```

Comment:
This would not be safe.
Maybe the bookie is up but it is, not doing what it is supposed to do

Reply:
Seems this is for handling issue of other components, which is not bookie. It is make sure the bookie still get running while other components meet exception.